### PR TITLE
Provide an endpoint to upgrade plugin

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/area core
/milestone 2.0

#### What this PR does / why we need it:

Provide an endpoint to upgrade plugin by uploading a new jar file.

#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/2551

#### Special notes for your reviewer:

1. Install an old plugin
2. Update the plugin and package again
3. Upgrade the plugin and see the result

#### Does this PR introduce a user-facing change?

```release-note
提供插件更新功能
```

### CODE REVIEW

**Review by ruibaby — APPROVED** [review]
/approve

File: src/main/java/run/halo/app/core/extension/endpoint/PluginEndpoint.java [hunk 1]
```diff
@@ -73,6 +82,16 @@ public RouterFunction<ServerResponse> endpoint() {
                         ))
                     .response(responseBuilder().implementation(Plugin.class))
             )
+            .POST("plugins/{name}/upgrade", contentType(MediaType.MULTIPART_FORM_DATA),
```
Comment (guqing) [inline src/main/java/run/halo/app/core/extension/endpoint/PluginEndpoint.java:85]: 这个 endpoint 需要在 `src/main/resources/extensions/role-template-plugin.yaml` 的 role-template-manage-plugins 角色模板下添加一下 rule

Comment (JohnNiang) [reply src/main/java/run/halo/app/core/extension/endpoint/PluginEndpoint.java:85]: Good catch!

**Review by guqing — COMMENTED** [review]
/lgtm

**Pull request merged**